### PR TITLE
Allow setting bond MTU

### PR DESCRIPTION
**Pull request by mmirecki**
Allow setting the bond MTU value.

### CODE REVIEW

**Comment by mmirecki** [comment]
@killianmuldoon @rkamudhan Could you please look at this one, thanks

File: README.md [hunk 1]
```diff
@@ -50,6 +51,7 @@ Given the following network configuration:
 		"type": "bond",
 		"mode": "active-backup",
 		"miimon": "100",
+    "mtu": 1500,
```
Comment (SchSeba) [inline README.md:54]: please fix the indentation here

File: bond/bond.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ type bondingConfig struct {
 	FailOverMac int                      `json:"failOverMac"`
 	Miimon      string                   `json:"miimon"`
 	Links       []map[string]interface{} `json:"links"`
+	MTU         string                   `json:"mtu"`
```
Comment (SchSeba) [inline bond/bond.go:45]: should this be int? in the doc you mark it as int

File: bond/bond.go [hunk 3]
```diff
@@ -102,18 +103,23 @@ func checkLinkExists(linkName string, netNsHandle *netlink.Handle) (netlink.Link
...
 	bondModeObj := netlink.StringToBondMode(bondMode)
 	bondLinkObj.Attrs().Name = bondName
 	bondLinkObj.Mode = bondModeObj
 	bondLinkObj.Miimon, err = strconv.Atoi(bondMiimon)
-	bondLinkObj.FailOverMac = netlink.BondFailOverMac(failOverMac)
-
 	if err != nil {
 		return nil, fmt.Errorf("Failed to convert bondMiimon value (%+v) to an int, error: %+v", bondMiimon, err)
 	}
+	if len(bondMTU) > 0 {
```
Comment (SchSeba) [inline bond/bond.go:116]: should we also validate that the MTU number provided is valid?

Comment (SchSeba) [inline bond/bond.go:116]: do we also want to validate that the VFs we are going to attach have the same/higher MTU before we do it?

File: README.md [hunk 4]
```diff
@@ -31,6 +31,7 @@ The binary should be placed at /opt/cni/bin on all nodes on which bonding will t
 - type (string, required): &quot;bond&quot;
 - ifname (string, optional): name of the bond interface
 - miimon (int, required): specifies the arp link monitoring frequency in milliseconds
+- mtu (int, optional): the mtu of the bond
```
Comment (cgoncalves) [inline README.md:34]:
```suggestion
- mtu (int, optional): the MTU of the bond. Default is 1500.
```

File: bond/bond.go (hunk 3 above)
Comment (mmirecki) [reply bond/bond.go]: Added validation that the mtu is not higher than the mtu of the pf of any of the vf's used. The link mac is used to match the pf's vf to the slave used in the bond.
The original vf mtu's are not relevant, as adding a link to a bond will modify it's mtu to that of the bond.

Comment (mmirecki) [reply bond/bond.go]: The VF mtu's will be automatically changed to that of the bond.

File: bond/bond.go (hunk 2 above)
Comment (mmirecki) [reply bond/bond.go]: done

File: README.md (hunk 4 above)
Comment (mmirecki) [reply README.md]: added

File: bond/bond.go [hunk 5]
```diff
@@ -225,6 +229,41 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
...
+	if err != nil {
+		return fmt.Errorf("Failed to create a new handle, error: %+v", err)
+	}
+	defer netHandle.Delete()
+
+	pfLinks, err := netHandle.LinkList()
+	if err != nil {
+		return 	fmt.Errorf("Failed to lookup physical functions links, error: %+v", err)
+	}
+	for _, pfLink := range pfLinks {
```
Comment (SchSeba) [inline bond/bond.go:249]: how you are able to get the PF info? you are inside the pod namespace already no?

Comment (mmirecki) [reply bond/bond.go:249]: Note that a cni does not work inside a namespace. You have to enter the namespace yourself.
One of the input parameters the CNI binary is getting is the CNI_NETS environment variable, which points us to the namespace for the container it is to connect.
We are getting the netlink handle for the pf without specifying the ns ([here](https://github.com/k8snetworkplumbingwg/bond-cni/blob/296681abbb01c9ab35c0f38af9a1769761d1e931/bond/bond.go#L239)), so we operate with no namespace here.
The slaves are retrieved in  getLinkObjectsFromConfig() which takes a handle created using a ns ([here](https://github.com/k8snetworkplumbingwg/bond-cni/blob/296681abbb01c9ab35c0f38af9a1769761d1e931/bond/bond.go#L278)).

**Review by Eoghan1232 — APPROVED** [review]

**Comment by cgoncalves** [comment]
@Eoghan1232 @SchSeba thanks for the review and approval. Is there anything else missing or can this PR be merged?

**Comment by Eoghan1232** [comment]
PR approved by myself and @SchSeba.
Proceeding with Merge.

**Pull request merged**